sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/CommunicationIdentityAsyncTests.java

```diff
@@ -92,6 +94,114 @@ public void createUserAndToken() {
             .verifyComplete();
     }
 
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("com.azure.communication.identity.TokenCustomExpirationTimeHelper#getValidExpirationTimes")
+    public void createUserAndTokenWithValidCustomExpiration(String testName, Duration tokenExpiresAfter) {
+        // Arrange
+        CommunicationIdentityClientBuilder builder = createClientBuilder(httpClient);
+        asyncClient = setupAsyncClient(builder, "createUserAndTokenWithValidCustomExpiration" + testName);
+        List<CommunicationTokenScope> scopes = Arrays.asList(CommunicationTokenScope.CHAT);
+
+        // Action & Assert
+        Mono<CommunicationUserIdentifierAndToken> createUserAndToken = asyncClient.createUserAndToken(scopes, tokenExpiresAfter);
+        StepVerifier.create(createUserAndToken)
+            .assertNext(result -> {
+                assertNotNull(result.getUserToken());
```

Comment:
should we also assert whether expiration time is as requested?(same for test cases where we'r testing for a valid param)

Reply:
True. I didn't check for specific time. Thought it would be a bit brittle, but saw Auth actually added tests for time difference. They're comparing them with token expiration. Should we do it also? @petrsvihlik wdyt?

Reply:
hmm...an interesting idea. I think it only makes sense for LIVE live tests as the recordings would always have to be adjusted to contain a correct value. in .NET, this is achievable by adding `[Category(TestCategory.Live)]` attribute to a test class or method. I expect, there'll be something similar for java.

we could allow for about 5% difference which should cover for any time skew even for the shortest possible expiration interval of 60 mins (=> 3mins tolerance).

Reply:
for java you can assert only if `getTestMode() == TestMode.LIVE`

Reply:
Done.